bitchat/Features/media/ImageUtils.swift

```diff
@@ -0,0 +1,149 @@
+#if os(iOS)
+import UIKit
+#else
+import AppKit
+import ImageIO
+import UniformTypeIdentifiers
+#endif
+
+enum ImageUtilsError: Error {
+    case invalidImage
+    case encodingFailed
+}
+
+enum ImageUtils {
+    private static let compressionQuality: CGFloat = 0.85
+    private static let targetImageBytes: Int = 60_000
+
+    static func processImage(at url: URL, maxDimension: CGFloat = 512) throws -> URL {
+        let data = try Data(contentsOf: url)
+        #if os(iOS)
+        guard let image = UIImage(data: data) else { throw ImageUtilsError.invalidImage }
+        return try processImage(image, maxDimension: maxDimension)
+        #else
+        guard let image = NSImage(data: data) else { throw ImageUtilsError.invalidImage }
+        return try processImage(image, maxDimension: maxDimension)
+        #endif
+    }
+
+    #if os(iOS)
+    static func processImage(_ image: UIImage, maxDimension: CGFloat = 512) throws -> URL {
+        let scaled = scaledImage(image, maxDimension: maxDimension)
+        var quality = compressionQuality
+        guard var jpegData = scaled.jpegData(compressionQuality: quality) else {
+            throw ImageUtilsError.encodingFailed
+        }
+        while jpegData.count > targetImageBytes && quality > 0.3 {
+            quality -= 0.1
+            guard let next = scaled.jpegData(compressionQuality: quality) else { break }
+            jpegData = next
+        }
+        let outputURL = try makeOutputURL()
+        try jpegData.write(to: outputURL, options: .atomic)
+        return outputURL
+    }
+
+    private static func scaledImage(_ image: UIImage, maxDimension: CGFloat) -> UIImage {
+        let size = image.size
+        let maxSide = max(size.width, size.height)
+        guard maxSide > maxDimension else { return image }
+        let scale = maxDimension / maxSide
+        let newSize = CGSize(width: size.width * scale, height: size.height * scale)
+        UIGraphicsBeginImageContextWithOptions(newSize, true, 1.0)
+        image.draw(in: CGRect(origin: .zero, size: newSize))
+        let rendered = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return rendered ?? image
+    }
+    #else
+    static func processImage(_ image: NSImage, maxDimension: CGFloat = 512) throws -> URL {
+        let scaled = scaledImage(image, maxDimension: maxDimension)
+        guard let inputCG = scaled.cgImage(forProposedRect: nil, context: nil, hints: nil) else {
+            throw ImageUtilsError.encodingFailed
+        }
+        let width = inputCG.width
+        let height = inputCG.height
+        let colorSpace = CGColorSpace(name: CGColorSpace.sRGB) ?? CGColorSpaceCreateDeviceRGB()
+        guard let context = CGContext(
+            data: nil,
+            width: width,
+            height: height,
+            bitsPerComponent: 8,
+            bytesPerRow: 0,
+            space: colorSpace,
+            bitmapInfo: CGImageAlphaInfo.premultipliedLast.rawValue
```

Comment:
I think this approach works for both platforms and could simplify the code by unifying the code